dispatcher/main.go

```diff
@@ -52,170 +47,66 @@ func main() {
 	run, err := models.QueryRun(*pRunID)
 	common.ExitOnError(err, "fail to query the run")
 
-	run.Details[common.KeyProduct] = droidMetadata.Product
-	run, err = run.Patch()
-	common.ExitOnError(err, "fail to update the run")
-
-	printRunInfo(run)
+	if run.Status == common.RunStatusInitialized || len(run.Status) == 0 {
```

Comment:
>if [](start = 1, length = 2)

No switch? :D

Reply:
Wait no, I see you need all this in order, ignore my comment.

---
In reply to: [174318488](https://github.com/Azure/adx-automation-agent/pull/8#discussion_r174318488) [](ancestors = 174318488)